docs/proposals/source-verificiation-policies.md

```diff
@@ -0,0 +1,313 @@
+---
+title: Source Verification Policies
+authors:
+  - "@jannfis" # Authors' github accounts here.
+sponsors:
+  - TBD        # List all interested parties here.
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+creation-date: 2023-08-08
+last-updated: 2023-08-08
+---
+
+# Source Verification Policies
+
+This proposal introduces an evolution to the existing GnuPG commit verification feature.
+
+## Open Questions [optional]
+
+* Do we really need `progressive` verification level (see below)?
+* Should repository pattern matching be regexp based instead of shell-glob?
+
+## Summary
+
+Argo CD has had a feature to verify OpenPGP signatures on Git commits using GnuPG for a while. However, this feature is limited and in some ways may behave unexpectedly. Argo CD would ever only verify the signature on the commit that's pointed to by the `HEAD` of the `Application`'s resolved `targetRevision`. This allows for unsigned commits to slip in, and requires thorough verification of the source by the user prior to signing a commit.
+
+Source verification policies are an evolution of the legacy signature verification in Argo CD. A source verification policy defines the cryptographic requirements to be met in order for a source to be allowed to sync, and with it, brings new verification levels, the possibility of treating multiple sources in an Application with different policies and it sets the foundation for the ability to implement more verification methods in the future.
+
+## Motivation
+
+As a deployment tool, Argo CD sits in a prominent spot to enforce secure supply chain requirements. The cryptographic verification of sources and artifacts become more and more important to organisations and are heavily regulated in some fields.
+
+While having Argo CD verifying the `HEAD` commit to be synced is of course better than nothing, it is not a sufficient level of verification for many organisations and individuals anymore.
+
+Also, the current approach is an "all-or-nothing" approach for any given AppProject. The reality however is, that different repositories might have different trust levels, and different people working on them and being trusted. Especially with the advent of multi-source applications, the current approach of defining the trusted signers for all Applications in a given AppProject is not viable anymore.
+
+### Goals
```

Comment:
Should there be an additional goal of support GPG in ApplicationSet git generator?

Reply:
IMHO, ApplicationSet is out of scope for this proposal. ApplicationSets are not bound to AppProjects afaik.